schema_cache.rst

```diff
@@ -0,0 +1,131 @@
+| :ref:`open-api`                            | Table columns, primary keys and foreign keys                                  |
++                                            +-------------------------------------------------------------------------------+
+|                                            | View columns and INSTEAD OF TRIGGERS                                          |
++                                            +-------------------------------------------------------------------------------+
+|                                            | Function's metadata                                                           |
++--------------------------------------------+-------------------------------------------------------------------------------+
+
+The Stale Schema Cache
+----------------------
+
+When you make changes on the metadata mentioned above, the schema cache will turn stale on a running PostgREST. Future requests that use the above features will need the :ref:`schema to be reloaded <schema_reloading>`; otherwise, you'll get an error instead of the expected result.
+
+For instance, let's see what would happen if you have a stale schema for foreign key relationships and function metadata:
+
+Stale Foreign Key Relationships
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Suppose you add a ``cities`` table to your database. This table has a foreign key referencing an existing ``countries`` table. Then, you make a request to get the ``cities`` and their belonging ``countries``:
+
+.. code-block:: http
+
+  GET /cities?select=name,country:countries(id,name) HTTP/1.1
+
+But instead, you get an error message that looks like this:
+
+.. code-block:: json
+
+  {
+    "hint": "If a new foreign key between these entities was created in the database, try reloading the schema cache.",
+    "message": "Could not find a relationship between cities and countries in the schema cache"
+  }
+
+As you can see, PostgREST couldn't find the newly created foreign key in the schema cache. See the section :ref:`schema_reloading` to solve this issue.
+
+Stale Function Signature
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Suppose you create the following function while PostgREST is running:
+
+.. code-block:: plpgsql
+
+  CREATE FUNCTION plus_one(num integer)
+  RETURNS integer AS $$
+   SELECT num + 1;
+  $$ LANGUAGE SQL IMMUTABLE;
+
+Then, you make this request:
+
+.. code-block:: http
+
+  GET /rpc/plus_one?num=1 HTTP/1.1
+
+On a stale schema, PostgREST will assume :code:`text` as the default type for the function argument ``num``. Thus, the response you get is:
+
+.. code-block:: json
+
+ {
+  "hint":"No function matches the given name and argument types. You might need to add explicit type casts.",
+  "details":null,
+  "code":"42883",
+  "message":"function test.plus_one(num => text) does not exist"
+ }
+
+See the section :ref:`schema_reloading` to solve this issue.
+
+.. _schema_reloading:
+
+Schema Reloading
```

Comment:
I think the link should stay that way because we've referenced it many times in answers to users' issues. I don't think it's that inaccurate as a reference name either.

Reply:
Yeah, you're right. Too late to change it. The name is not so bad, let's leave it like that.